### DISCUSSION TITLE
Towards full-page sheet music

### DISCUSSION THREAD

**Discussion by suntong** [alda-lang/alda#473] [Ideas]
I've read and known that there is still a long way to go towards the end goal of high-quality sheet music export, 

one of the steps would be to export as [MusicXML](https://w3c.github.io/musicxml/), as in #297.

and I believe there is a one more tiny step needed before that -- to allow multi-line support in script. I.e., 

Instead of writing super long notes in a single line, better make it possible for people to write multi-line notes, like

```
# line 1
V1: c8 d e f g a b > c1
V2: e8 f g a b > c d e1
V3: g8 a b > c d e f g1
V4: o3 c2 g4 < g8 c1

# line 2
V1: c8 d e f g a b > c1
V2: e8 f g a b > c d e1
V3: g8 a b > c d e f g1
V4: o3 c2 g4 < g8 c1

# line 3
#...

```

Is such feature already implemented?

**Comment by daveyarwood** [comment]
Hi @suntong :wave: 

I might be misunderstanding your idea, but I believe Alda already works the way you're describing. There are some examples [here](https://github.com/alda-lang/alda/blob/master/doc/scores-and-parts.md) that might be helpful in understanding how parts work. Voices work very similarly.

**Comment by suntong** [comment]
Ok, I'll take your words for it, as I was trying on an online site, where there is a huge gap between line 1 & line 2. 

Maybe there'll be no (excessive) gaps when trying locally. thanks.

**Reply by daveyarwood** [reply]
Was it Severak's [Alda web workspace](https://severak.github.io/alda-js/)? I tried the same code example above (adding `piano:` at the top) there and it also sounded the way that I expect. Each voice ends with a whole note, which is a long note. Maybe that's the gap you were hearing? Just trying to make sure I'm understanding the issue correctly :)

**Reply by suntong** [reply]
> Each voice ends with a whole note, which is a long note.

Ah~~!! That's why! Thanks. 

Yes, this is what I tried just now, 

```
piano:

# line 1
V1: c8 d e f g a b > c <
V2: e8 f g a b > c d e
V3: g8 a b > c d e f g
V4: o3 c2 g4 < g8 c >

# line 2
V1: c8 d e f g a b > c1
V2: e8 f g a b > c d e1
V3: g8 a b > c d e f g1
V4: o3 c2 g4 < g8 c1
```

And it sounds the way that I expect now. 

Thanks

**Reply by daveyarwood** [reply]
Great - happy to help!
